lib/maths/CMathsFuncs.cc

```diff
@@ -8,7 +8,7 @@
 
 #include <maths/CMathsFuncsForMatrixAndVectorTypes.h>
 
-#include <boost/math/special_functions/fpclassify.hpp>
+#include <cmath>
 
 #ifdef isnan
```

Comment:
If this macro is still needed then `std::isfinite` should probably be in brackets in other places where it's used.  Otherwise the check on this macro and `isinf` immediately below can be removed.

Reply:
I checked this. We are not including math.h - the C header file - anymore, which should avoid this problem. But even then, according to this blog post we should be fine:

https://developers.redhat.com/blog/2016/02/29/why-cstdlib-is-more-complicated-than-you-might-think/

FWIW: Prior to this PR I build this code on both `master` and `6.x`, so at least on these platforms (afaik that includes Fedora) we should be fine.

I will remove these guards and re-run builds.

Reply:
> We are not including math.h - the C header file - anymore, which should avoid this problem.

Even if we're not including math.h from our code, we could be including a 3rd party header that includes it.

But that blog you linked is a good find because it contains the reason why the possibility of clashing macros is completely removed since we upgraded to GCC 6:

> For GCC 6 this has been fixed, by making the C++ library install its own version of <math.h>, which includes <cmath>, which then includes the real <math.h> from the C library.

So even if a 3rd party header includes math.h it will pick up the C++ versions of all the maths functions.  And since the fix is in GCC rather than the OS headers we're safe regardless of which Linux distribution we're building on, which is also nice.